### ISSUE TITLE
[Android] EventSink is null when use in conjunction with audio_service

### ISSUE THREAD

**Issue by Camerash** [ryanheise/just_audio#57]
- just_audio version: 0.1.3
- Crash log:
```
java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.EventChannel$EventSink.success(java.lang.Object)' on a null object reference
E/AndroidRuntime(20897): 	at com.ryanheise.just_audio.AudioPlayer.broadcastPlaybackEvent(AudioPlayer.java:229)
E/AndroidRuntime(20897): 	at com.ryanheise.just_audio.AudioPlayer.access$300(AudioPlayer.java:32)
E/AndroidRuntime(20897): 	at com.ryanheise.just_audio.AudioPlayer$1.run(AudioPlayer.java:65)
E/AndroidRuntime(20897): 	at android.os.Handler.handleCallback(Handler.java:883)
E/AndroidRuntime(20897): 	at android.os.Handler.dispatchMessage(Handler.java:100)
E/AndroidRuntime(20897): 	at android.os.Looper.loop(Looper.java:214)
E/AndroidRuntime(20897): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E/AndroidRuntime(20897): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(20897): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/AndroidRuntime(20897): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

This is possibly related to #34, and I believe a fix would be similar

**Comment by ryanheise** [comment]
Thanks for the report. Do you have steps to reproduce? I did not notice this with the sample.

**Comment by Camerash** [comment]
This is my current code on audio_service's onStop method:
```
@override
  void onStop() async {
    await this._recordingAudioPlayer.stop();
    await this._recordingAudioPlayer.dispose();
    await this._setState(state: BasicPlaybackState.stopped);
    this._audioCompleter.complete();
  }
```

After the completer completes, the `onStart` method resumes and cancels any active subscriptions on just_audio streams
This crash only happens when I call `stop` on audio service.

**Comment by ryanheise** [comment]
Would you mind testing what happens if you go to `AudioPlayer.java:229` and put a null check around that line? i.e. On line 229, change this:

```java
		eventSink.success(event);
```

to this:

```java
		if (eventSink != null)
			eventSink.success(event);
```

This should make the error go away, but to know if it completely works, you would need to see whether the eventSink is eventually set and events eventually start arriving in the UI. You'll know if the UI updates itself in response to these events after execute your reproduction steps.

**Comment by ryanheise** [comment]
Comment from @duncan-iaria on #61:

> This seems to have resolved the issue for me. Thank you! Btw that other issue mentions using audio_service as well - in my case I was only using just_audio, not sure if that is helpful information or not but I thought I'd throw it out there.

Thanks for confirming @duncan-iaria , I've just committed the fix and will hopefully include it in the next release within the next day.

**Comment by ryanheise** [comment]
Included in release 0.1.4.
